src/test/java/com/google/firebase/auth/FirebaseAuthIT.java

```diff
@@ -1066,6 +1066,137 @@ public void testTenantAwareOidcProviderConfigLifecycle() throws Exception {
     }
   }
 
+  @Test
+  public void testListOidcProviderConfigs() throws Exception {
+    final List<String> providerIds = new ArrayList<>();
+
+    try {
+      // Create provider configs
+      for (int i = 0; i < 3; i++) {
+        String providerId = "oidc.provider-id" + i;
+        providerIds.add(providerId);
+        OidcProviderConfig.CreateRequest createRequest = new OidcProviderConfig.CreateRequest()
+            .setProviderId(providerId)
+            .setClientId("CLIENT_ID")
+            .setIssuer("https://oidc.com/issuer");
+        auth.createOidcProviderConfig(createRequest);
+      }
+
+      // Test list by batches
+      final AtomicInteger collected = new AtomicInteger(0);
+      ListProviderConfigsPage<OidcProviderConfig> page =
+          auth.listOidcProviderConfigsAsync(null).get();
+      while (page != null) {
+        for (OidcProviderConfig providerConfig : page.getValues()) {
+          if (providerIds.contains(providerConfig.getProviderId())) {
+            collected.incrementAndGet();
+            assertEquals("CLIENT_ID", providerConfig.getClientId());
+            assertEquals("https://oidc.com/issuer", providerConfig.getIssuer());
+          }
+        }
+        page = page.getNextPage();
+      }
+      assertEquals(providerIds.size(), collected.get());
+
+      // Test iterate all
+      collected.set(0);
+      page = auth.listOidcProviderConfigsAsync(null).get();
+      for (OidcProviderConfig providerConfig : page.iterateAll()) {
+        if (providerIds.contains(providerConfig.getProviderId())) {
+          collected.incrementAndGet();
+          assertEquals("CLIENT_ID", providerConfig.getClientId());
```

Comment:
Define a helper to reduce some of this duplication.

```
if (checkProviderConfig(providerIds, providerConfig) {
  collected.incrementAndGet();
}

private boolean checkProviderConfig(List<String> providerIds, OidcProviderConfig config) {
  
}
```

Reply:
This doesn't save too many lines, but sure.